### PR TITLE
Make archive creation deterministic

### PR DESCRIPTION
**Pull request by StefanBruens**
This should address issue #101.

### CODE REVIEW

File: libappstream-builder/asb-utils.c [hunk 1]
```diff
@@ -441,8 +441,10 @@ asb_utils_write_archive (const gchar *filename,
 	struct stat st;
 
 	a = archive_write_new ();
-	if (g_str_has_suffix (filename, ".gz"))
+	if (g_str_has_suffix (filename, ".gz")) {
 		archive_write_add_filter_gzip (a);
+                archive_write_set_filter_option (a, "gz", "timestamp", NULL);
```
Comment (ignatenkobrain) [inline libappstream-builder/asb-utils.c]: style: please use tabs

File: libappstream-builder/asb-utils.c [hunk 2]
```diff
@@ -532,6 +534,9 @@ asb_utils_write_archive_dir (const gchar *filename,
 	if (files->len == 0)
 		return TRUE;
 
+        /* sort by filename for deterministic results */
```
Comment (ignatenkobrain) [inline libappstream-builder/asb-utils.c]: style: please use tabs (for 2 next lines as well)

File: libappstream-builder/asb-utils.c [hunk 3]
```diff
@@ -441,8 +441,10 @@ asb_utils_write_archive (const gchar *filename,
 	struct stat st;
 
 	a = archive_write_new ();
-	if (g_str_has_suffix (filename, ".gz"))
+	if (g_str_has_suffix (filename, ".gz")) {
 		archive_write_add_filter_gzip (a);
+                archive_write_set_filter_option (a, "gz", "timestamp", NULL);
+	}
 	if (g_str_has_suffix (filename, ".bz2"))
 		archive_write_add_filter_bzip2 (a);
```
Comment (ignatenkobrain) [inline libappstream-builder/asb-utils.c]: what about bzip/xz/etc.? Don't we need to do the same for those?

Comment (StefanBruens) [inline libappstream-builder/asb-utils.c]: At least in libarchive the bz2 and xz filter have no timestamp option. *if* these formats have a timestamp(but I think they don't), these filters had to be enhanced in libarchive first.

Setting these options anyway would create an warning in libarchive.

File: libappstream-builder/asb-utils.c (hunk 2 above)
Comment (StefanBruens) [inline libappstream-builder/asb-utils.c]: fixed

**Comment by ignatenkobrain** [comment]
@StefanBruens thanks, merged!

**Pull request merged**